src/Accelerators/Accelerator.hpp

```diff
@@ -28,7 +28,10 @@
 // generated file Accelerators.inc).
 #define CREATE_ACCEL_ENUM(name) name,
 #define DECLARE_ACCEL_INIT_FUNCTION(name) extern Accelerator *create##name();
-#define INVOKE_ACCEL_INIT_FUNCTION(name) create##name()->setName(#name);
+#define INVOKE_ACCEL_INIT_FUNCTION(name, kinds)                                \
+  if (kinds.empty() ||                                                         \
```

Comment:
> I don't understand why we support empty kinds (which is maccel) differently?

`onnx-mlir-opt` will initialize all accelerators so that accelerators' passes will be available. If you prefer, we can allow `-maccel` in `onnx-mlir-opt` so that only passes for the given accelerators are available, and empy kind means there is no accelerator.

> I would rather allow building for NNPA on a non-Z machine and do a lit test on a x86 machine by expressing the -maccel=NNPA in the onnx-mlir or onnx-mlir-opt.

This is another thing. It depends on whether we build libzdnn or not. If not, we can do the lit test on a x86 machine. A PR that allows turning on/off libzdnn build can solve the problem.

Reply:
> There is something funny about the lit-tests, and maybe that is the reason of the confusion? In the makefiles, we disable building for NNPA if we are not on an Z machine... and maybe that is the source of the problem? I would rather allow building for NNPA on a non-Z machine and do a lit test on a x86 machine by expressing the `-maccel=NNPA` in the `onnx-mlir` or `onnx-mlir-opt`.

There is no reason to build NNPA on non-Z machines since it will never run there, and that's why PR #1384 was created to allow NNPA lit tests on non-Z machines even if NNPA cannot be built.

Reply:
>  If you prefer, we can allow -maccel in onnx-mlir-opt so that only passes for the given accelerators are available, and empy kind means there is no accelerator.

yes, that would be my preference. That way things looks more like `onnx-mlir`.